Sources/SATSCore/Extensions/SwiftUI/ViewData/AlertViewData.swift

```diff
@@ -43,21 +52,43 @@ public struct AlertViewData: Identifiable, Equatable {
 public extension Alert {
     /// Creates an alert from view data, this include a default cancel button
     init(viewData: AlertViewData) {
-        let primaryButton: Alert.Button
-
-        switch viewData.actionButtonStyle {
-        case .default:
-            primaryButton = .default(Text(viewData.actionTitle), action: viewData.action)
+        let primaryButton = Self.createActionButton(
+            style: viewData.actionButtonStyle,
+            title: viewData.actionTitle,
+            action: viewData.action
+        )
 
-        case .destructive:
-            primaryButton = .destructive(Text(viewData.actionTitle), action: viewData.action)
-        }
+        let secondaryButton = Self.createActionButton(
+            style: viewData.secondaryButtonStyle,
+            title: viewData.secondaryActionTitle,
+            action: viewData.secondaryAction
+        )
```

Comment:
I think it's better to do something like
```suggestion
        let secondaryButton: Alert.Button
        if let secondaryActionTitle = viewData.secondaryActionTitle, let secondaryAction = viewData.secondaryAction {
            secondaryButton = Self.createActionButton(
                style: viewData.secondaryButtonStyle,
                title: secondaryActionTitle,
                action: secondaryAction
            )
        } else {
            secondaryButton = .cancel()
        }
```

the `createActionButton` as is right now assumes that if any property is missing, you are building a `cancel` button

that's too much of an assumption IMO.

then it's better to localize that behavior to the secondary button alone instead.

Reply:
Agree!